Odd situation using Multibyte keys

`len` returns the number of bytes of string, not number of runes. 
example: `len("こんにちは")` returns 15, not 5. https://play.golang.org/p/ysUcFI-0YhL

If you use CommonPrefixSearch with multibyte keys, member `Key` in return value contains corrupted key.

I wrote some test. It looks it's not right.

```go

func TestCommonPrefixSearch_Multibyte(t *testing.T) {
	d := New()
	d.Write("こんにちは", "hello")
	d.Write("さようなら", "bye")

	test := struct {
		in  string
		out CommonPrefixSearchResult
	}{
		"おはようこんにちはさようなら",
		[]struct {
			Pos   int
			Key   string
			Value interface{}
		}{
			{4, "こんにちは", "hello"},
			{9, "さようなら", "bye"},
		},
	}

	t.Run(test.in, func(t *testing.T) {
		ok := true
		result := d.CommonPrefixSearch(test.in)

		for i, r := range result {
			if test.out[i] != r {
				ok = false
			}
		}

		if !ok {
			t.Errorf("got %v, want %v", result, test.out)
		}
	})
}
```

```console
=== RUN   TestCommonPrefixSearch_Multibyte/おはようこんにちはさようなら
--- FAIL: TestCommonPrefixSearch_Multibyte (0.00s)
    --- FAIL: TestCommonPrefixSearch_Multibyte/おはようこんにちはさようなら (0.00s)
        damap_test.go:124: got [{12 こんにち hello} {27 さような bye}], want [{4 こんにちは hello} {9 さようなら bye}]
```

`こんにち`, `さような` are corrupted.

--- a/damap.go
+++ b/damap.go
@@ -1,6 +1,8 @@
 // Package damap provides a map data structure on double-array trie tree and functions to search values from trie tree.
 package damap
 
+import "unicode/utf8"
+
 const endKey rune = 0
 
 // DaMap is a map data structure based on double-array trie tree.
@@ -187,16 +189,17 @@ type CommonPrefixSearchResult []struct {
 }
 
 // CommonPrefixSearch provides common prefix search functions for trie tree.
-func (d *DaMap) CommonPrefixSearch(s string) CommonPrefixSearchResult {
-	result := CommonPrefixSearchResult{}
-
-	if len(s) == 0 {
-		return result
+func (d *DaMap) CommonPrefixSearch(s string) (r CommonPrefixSearchResult) {
+	rl := utf8.RuneCountInString(s)
+	if rl == 0 {
+		return
 	}
 
-	for pos := 0; pos < len(s); pos++ {
+    rs := []rune(s)
+
+	for pos := 0; pos < rl; pos++ {
 		i := 0
-		for j, k := range s[pos:] {
+		for j, k := range rs[pos:] {
 			n := d.base[i] + int(k) - 1
 
 			if len(d.base) < n+1 {
@@ -209,15 +212,15 @@ func (d *DaMap) CommonPrefixSearch(s string) CommonPrefixSearchResult {
 
 			leaf := d.base[n] + int(endKey) - 1
 			if d.base[leaf] == 0 && d.check[leaf] == n+1 {
-				result = append(
-					result,
+				r = append(
+					r,
 					struct {
 						Pos   int
 						Key   string
 						Value interface{}
 					}{
 						Pos:   pos,
-						Key:   string(s[pos : pos+j+1]),
+						Key:   string(rs[pos : pos+j+1]),
 						Value: d.value[leaf],
 					},
 				)
@@ -227,5 +230,5 @@ func (d *DaMap) CommonPrefixSearch(s string) CommonPrefixSearchResult {
 		}
 	}
 
-	return result
+	return
 }

--- a/damap_test.go
+++ b/damap_test.go
@@ -83,3 +83,32 @@ func TestCommonPrefixSearch(t *testing.T) {
 		}
 	})
 }
+
+func TestMultiByteTextCommonPrefixSearch(t *testing.T) {
+	d := New()
+	d.Write("こんにちは", "hello")
+	d.Write("さようなら", "bye")
+
+	test := struct {
+		in  string
+		out CommonPrefixSearchResult
+	}{
+		"おはようこんにちはさようなら",
+		[]struct {
+			Pos   int
+			Key   string
+			Value interface{}
+		}{
+			{4, "こんにちは", "hello"},
+			{9, "さようなら", "bye"},
+		},
+	}
+
+	t.Run(test.in, func(t *testing.T) {
+		result := d.CommonPrefixSearch(test.in)
+
+		if !reflect.DeepEqual(result, test.out) {
+			t.Errorf("got %v, want %v", result, test.out)
+		}
+	})
+}

